File: src/main/lesson7/javahw/Role.java
package javahw;

public enum Role {
    ADMIN("1cvn_admin_token"),
    MODERATOR("2cvm_mod_token"),
    USER("3cgh_user_token");

    private final String token;

    Role(String token) {
        this.token = token;
    }

    public boolean customEquals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        return token.equals(((Role) o).token);
    }

}

File: src/main/lesson7/kotlinhw/Role.kt
package kotlinhw

enum class Role(val token: String) {

    ADMIN("1cvn_admin_token"),
    MODERATOR("3cvm_mod_token"),
    USER("2cvn_admin_token");

    fun customEquals(other: Any?): Boolean {
        if (this == other) return true
        if (javaClass != other?.javaClass) return false

        other as Role
        return token == other.token
    }
}